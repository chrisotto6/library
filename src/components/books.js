import React, { useState, useEffect } from 'react'
import xml2js from 'xml2js'
import Card from './card'
import './books.styles.css'

const Books = () => {
  const [books, setBooks] = useState([])
  const [isLoading, setIsLoading] = useState(true)

  useEffect(() => {
    fetch(
      `https://cors-anywhere.herokuapp.com/https://www.goodreads.com/review/list?v=2&id=96692540&shelf=read&sort=date_read&per_page=200&key=${process.env.REACT_APP_API_KEY}`,
      {
        method: 'GET',
      }
    )
      .then((res) => res.text())
      .then((response) => {
        xml2js.parseString(response, (err, res) => {
          if (err) console.log(err)
          setBooks(res.GoodreadsResponse.reviews[0].review)
          setIsLoading(false)
        })
      })
      .catch((error) => console.log(error))
  }, [])
  return (
    <div className='booksContainer'>
      {isLoading && (
        <p style={{ color: 'black' }}>Loading books from Goodreads...</p>
      )}
      <ul>
        {!isLoading &&
          books.map((book, index) => (
            <Card book={book.book[0]} rating={book.rating[0]} key={index} />
          ))}
      </ul>
    </div>
  )
}

export default Books
