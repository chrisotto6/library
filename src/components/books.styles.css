ul {
  list-style-type: none;
  display: flex;
  flex-wrap: wrap;
  padding: 10px;
}

.booksContainer {
  justify-content: center;
  align-items: center;
  text-align: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-content: center;
}

.loading {
  color: black;
  text-align: center;
  padding: 50px;
  font-size: 24px;
}
