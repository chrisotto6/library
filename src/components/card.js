import React from 'react'
import StarRatingComponent from 'react-star-rating-component'

import './card.styles.css'

const Card = (props) => {
  const {
    book: { title, title_without_series, authors, image_url, link },
    rating,
  } = props

  return (
    <>
      <li className="card">
        <header>
          <div>
            <img src={image_url} alt={title} />
          </div>
        </header>
        <section className="container">
          <h4>{title_without_series}</h4>
          <h5>{authors.author.name}</h5>
          <StarRatingComponent name={''} value={Number(rating)} editing={false} />
          <div className="link">
            <a href={link} target="_blank" rel="noreferrer">
              Goodreads Page
            </a>
          </div>
        </section>
      </li>
    </>
  )
}

export default Card
