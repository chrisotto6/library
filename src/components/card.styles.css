.card {
  /* Add shadows to create the "card" effect */
  box-shadow: 0 4px 8px 0 white;
  flex: 5 5 calc(25% - 10px);
  display: flex;
  list-style: none;
  flex-direction: column;
  transition: 0.3s;
  text-align: center;
  justify-content: center;
  background-color: white;
  max-width: 100%;
  margin: 5px;
}

/* On mouse-over, add a deeper shadow */
.card:hover {
  box-shadow: 0 4px 10px 0 #282c34;
}

header {
  padding: 10px;
  position: relative;
}

/* Add some padding inside the card container */
.container {
  padding: 2px 16px;
}

h4,
h5 {
  padding: 5px 0 5px 0;
  margin: 5px 0;
}

.link {
  padding: 10px 0 10px 0;
}
