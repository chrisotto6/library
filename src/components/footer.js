import React from 'react'

function Footer() {
  const year = new Date().getFullYear()
  return (
    <div className="footer">
      <footer class="py-4 navbar-dark bg-secondary">
        <div class="container">
          <p class="m-0 text-center text-white">Copyright &copy; Chris Otto {year}</p>
        </div>
      </footer>
    </div>
  )
}

export default Footer
