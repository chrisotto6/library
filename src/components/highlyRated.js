import React from 'react'
import useFetch from '../hooks/useFetch'
import Card from './card'
import './bookList.styles.css'

const HighlyRated = () => {
  const url = `https://cotto-cors.herokuapp.com/https://www.goodreads.com/review/list?v=2&id=96692540&shelf=read&per_page=100&sort=rating&key=${process.env.REACT_APP_API_KEY}`
  const [data, isLoading] = useFetch(url, [])
  console.log(data)
  return (
    <div className="booksContainer">
      {isLoading && <p className="loading">Loading books from Goodreads...</p>}
      <ul>
        {!isLoading && data.map((book, index) => <Card book={book.book} rating={book.rating} key={index} />)}
      </ul>
    </div>
  )
}

export default HighlyRated
