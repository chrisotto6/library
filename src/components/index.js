export { default as RecentBooks } from './recentBooks'
export { default as Navigation } from './navigation'
export { default as Footer } from './footer'
export { default as Card } from './card'
export { default as ToReadBooks } from './toReadBooks'
export { default as HighlyRated } from './highlyRated'
