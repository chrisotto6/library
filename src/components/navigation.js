import React from 'react'
import { Link, withRouter } from 'react-router-dom'

const Navigation = (props) => {
  return (
    <div className="navigation">
      <nav className="navbar navbar-expand-sm navbar-dark bg-secondary justify-content-between">
        <div className="container-fluid">
          <Link className="navbar-brand" to="/">
            Library
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarResponsive"
            aria-controls="navbarResponsive"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarResponsive">
            <ul className="navbar-nav ml-auto">
              <li className={`nav-item  ${props.location.pathname === '/' ? 'active' : ''}`}>
                <Link className="nav-link" to="/">
                  Recently Read
                  <span className="sr-only">(current)</span>
                </Link>
              </li>
              <li className={`nav-item  ${props.location.pathname === '/to-read' ? 'active' : ''}`}>
                <Link className="nav-link" to="/to-read">
                  To Read
                </Link>
              </li>
              <li className={`nav-item  ${props.location.pathname === '/highly-rated' ? 'active' : ''}`}>
                <Link className="nav-link" to="/highly-rated">
                  Highly Rated
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  )
}

export default withRouter(Navigation)
