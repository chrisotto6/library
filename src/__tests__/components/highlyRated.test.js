import React from 'react'
import renderer from 'react-test-renderer'
import { HighlyRated } from '../../components'

test('HighlyRated Snapshot Matches', () => {
  const tree = renderer.create(<HighlyRated />).toJSON()
  expect(tree).toMatchSnapshot()
})
