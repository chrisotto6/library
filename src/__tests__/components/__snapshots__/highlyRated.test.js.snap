// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlyRated Snapshot Matches 1`] = `
<div
  className="booksContainer"
>
  <p
    className="loading"
  >
    Loading books from Goodreads...
  </p>
  <ul />
</div>
`;
